core/src/main/java/com/gorisse/thomas/sceneform/arcore/LightEstimate.kt

```diff
@@ -0,0 +1,263 @@
+                sphericalHarmonic *
+                        // Convert Environmental HDR's spherical harmonics to Filament
+                        // irradiance spherical harmonics.
+                        // TODO: Should we apply the cameraExposureFactor?
+                        Environment.SPHERICAL_HARMONICS_IRRADIANCE_FACTORS[index / 3]
+            }.toFloatArray()
+        }
+    } else {
+        baseEnvironment?.sphericalHarmonics
+    },
+    intensity = baseEnvironment?.indirectLight?.intensity)
+
+fun Frame.mainLightInfluenceEstimate(
+    baseLight: Light,
+    config: LightEstimationConfig,
+    // ARCore's light estimation uses unit-less (relative) values while Filament uses a physically
+    // based camera model with lux or lumen values.
+    // In order to keep the "standard" Filament behavior we scale AR Core values.
+    cameraExposureFactor: Float
+) = lightEstimate?.takeIf { it.state == LightEstimate.State.VALID }?.let { lightEstimate ->
+    when (config.mode) {
+        Config.LightEstimationMode.ENVIRONMENTAL_HDR ->
+            lightEstimate.environmentalHdrMainLightInfluence(
+                baseLight,
+                cameraExposureFactor,
+                config.environmentalHdrMainLightDirection,
+                config.environmentalHdrMainLightIntensity
+            )
+        else -> null
+    }
+}
+
+fun LightEstimate.environmentalHdrMainLightInfluence(
+    baseLight: Light,
+    // ARCore's light estimation uses unit-less (relative) values while Filament uses a physically
+    // based camera model with lux or lumen values.
+    // In order to keep the "standard" Filament behavior we scale AR Core values.
+    cameraExposureFactor: Float,
+    withDirection: Boolean = true,
+    withIntensity: Boolean = true
+): (Light.() -> Unit) = {
+    if (withDirection) {
+        environmentalHdrMainLightDirection.let { (x, y, z) ->
+            // TODO (or not):
+            //  If light is detected as shining up from below, we flip the Y
+            //  component so that we always end up with a shadow on the ground to
+            //  fulfill UX requirements.
+            direction = Direction(-x, -y, -z)
+        }
+    }
+    if (withIntensity) {
+        // Returns the intensity of the main directional light based on the inferred
+        // Environmental HDR Lighting Estimation. All return values are larger or equal to zero.
+        // The color correction method uses the green channel as reference baseline and scales the
+        // red and blue channels accordingly. In this way the overall intensity will not be
+        // significantly changed
+        environmentalHdrMainLightIntensity.let { (redIntensity, greenIntensity, blueIntensity) ->
+            // TODO : Do we have to apply the gray scale and/or convert to linear?
+            val colorIntensities = Float3(redIntensity, greenIntensity, blueIntensity) *
+                    cameraExposureFactor
```

Comment:
@devbridie Do you confirm that the color corrections are provided in linear space?